### PR TITLE
Add missing CopyableBuilder interfaces and fixed buggy ones.

### PR DESCRIPTION
**Pull request by millems**
Add CopyableBuilder and ToCopyableBuilder interfaces to credential providers. Also added spotbugs rule to validate that these and all other toBuilder methods are correct. Fixed any issues found with existing methods.

### CODE REVIEW

File: build-tools/src/main/java/software/amazon/awssdk/buildtools/findbugs/ToBuilderIsCorrect.java [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+
+            boolean result = false;
+            JavaClass[] parentInterfaces = obj.getAllInterfaces();
+            for (JavaClass i : parentInterfaces) {
+                if (i.getClassName().equals("software.amazon.awssdk.utils.builder.CopyableBuilder")) {
+                    builders.put(obj.getClassName(), obj);
+                    builderParents.put(obj.getClassName(),
+                                       Stream.concat(Arrays.stream(parentInterfaces), Arrays.stream(obj.getSuperClasses()))
+                                             .map(JavaClass::getClassName)
+                                             .collect(Collectors.toSet()));
```
Comment (cenedhryn) [inline build-tools/src/main/java/software/amazon/awssdk/buildtools/findbugs/ToBuilderIsCorrect.java:168]: It looks like the same code will get executed for any matching `i`. Is there a way to break it out of the loop?

File: core/annotations/src/main/java/software/amazon/awssdk/annotations/ToBuilderIgnoreField.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import java.lang.annotation.Target;
+
+/**
+ * Used to suppress certain fields from being considered in the spot-bugs rule for toBuilder(). This annotation must be
+ * attached to the toBuilder() method to function.
+ */
+@Target(ElementType.METHOD)
+@Retention(RetentionPolicy.CLASS)
+@SdkProtectedApi
+public @interface ToBuilderIgnoreField {
```
Comment (cenedhryn) [inline core/annotations/src/main/java/software/amazon/awssdk/annotations/ToBuilderIgnoreField.java:30]: Nice addition!

File: core/auth/src/main/java/software/amazon/awssdk/auth/credentials/ProcessCredentialsProvider.java [hunk 3]
```diff
@@ -83,9 +92,11 @@ private ProcessCredentialsProvider(Builder builder) {
 
         cmd.add(builderCommand);
 
-        this.command = Collections.unmodifiableList(cmd);
+        this.executableCommand = Collections.unmodifiableList(cmd);
```
Comment (cenedhryn) [inline core/auth/src/main/java/software/amazon/awssdk/auth/credentials/ProcessCredentialsProvider.java:95]: why was this renamed

Comment (millems) [reply core/auth/src/main/java/software/amazon/awssdk/auth/credentials/ProcessCredentialsProvider.java:95]: Mostly to differentiate it from the `commandFromBuilder`.

File: build-tools/src/main/java/software/amazon/awssdk/buildtools/findbugs/ToBuilderIsCorrect.java (hunk 1 above)
Comment (millems) [reply build-tools/src/main/java/software/amazon/awssdk/buildtools/findbugs/ToBuilderIsCorrect.java:168]: Nice catch, yes!

Comment (millems) [reply build-tools/src/main/java/software/amazon/awssdk/buildtools/findbugs/ToBuilderIsCorrect.java]: Technically it might be slower this way with it outside of the loop, since most classes won't have a matching value and even fewer will have multiple matching values.

**Review by cenedhryn — APPROVED** [review]
